pkgs/tools/package-management/nix/dependencies.nix

```diff
@@ -8,7 +8,18 @@ regular@{
 
 {
   scopeFunction = scope: {
-    boehmgc = regular.boehmgc.override { enableLargeConfig = true; };
+    boehmgc =
+      (regular.boehmgc.override {
+        enableLargeConfig = true;
+      }).overrideAttrs
+        {
+          # Increase the initial mark stack size to avoid stack overflows, since these inhibit
+          # parallel marking (see GC_mark_some()). To check whether the mark stack is too small,
+          # run Nix with GC_PRINT_STATS=1 and look for messages such as `Mark stack overflow`,
+          # `No room to copy back mark stack`, and `Grew mark stack to ... frames`.
+          NIX_CFLAGS_COMPILE = "-DINITIAL_MARK_STACK_SIZE=1048576";
```

Comment:
We should also run `nix-build -A nix.tests --keep-going` for safety to make sure we are not breaking stable nix.